src/layer/yolov3detectionoutput.h

```diff
@@ -39,6 +39,19 @@ class Yolov3DetectionOutput : public Layer
     Mat anchors_scale;
     int mask_group_num;
     ncnn::Layer* softmax;
+
+public:
+    struct BBoxRect
+    {
+        float xmin;
+        float ymin;
+        float xmax;
+        float ymax;
+        int label;
+    };
+    void qsort_descent_inplace(std::vector<BBoxRect>& datas, std::vector<float>& scores, int left, int right) const;
+    void qsort_descent_inplace(std::vector<BBoxRect>& datas, std::vector<float>& scores) const;
```

Comment:
The reason why the score was placed in a separate vector was to implement the bboxdata-independent template function. Here you can put the score member into BBoxRect struct to simplify the implementation of qsort.
It should be more memory efficient and fast.